test/factory.js

```diff
@@ -59,10 +59,48 @@ YUI(GlobalConfig).add('juju-tests-factory', function(Y) {
       return fakeStore;
     },
 
+    _fetchCharmData: function() {
```

Comment:
Sure, documentation added. This is used by the factory to fetch flat files which contain the api response json for a number of charms that various tests use.